src/matrixCells/makeMatrixArgs.js

```diff
@@ -5,24 +5,56 @@ import make_position_arr from "./makePositionArr";
 export default function make_matrix_args(regl, store) {
   // make arrays
   const arrs = {};
+  const row_nodes = store.select("network").row_nodes;
+  const col_nodes = store.select("network").col_nodes;
+  const {
+    highlighted_cols,
+    highlighted_rows,
+  } = store.select("search");
   arrs.opacity_arr = make_opacity_arr(store);
-  arrs.position_arr = {};
-  arrs.position_arr.ini = make_position_arr(
+  arrs.mask_arr = new Array(arrs.opacity_arr.length).fill(0.0);
+
+  const ini_dict = make_position_arr(
     store,
     store.select("order").inst.row,
     store.select("order").inst.col
   );
-  arrs.position_arr.new = make_position_arr(
+  const new_dict = make_position_arr(
     store,
     store.select("order").new.row,
     store.select("order").new.col
   );
-  store.dispatch(store.actions.setArrsState(arrs));
+  arrs.position_arr = {};
+  arrs.position_arr.ini = Object.values(ini_dict);
+  arrs.position_arr.new = Object.values(new_dict);
+
+  for (let i = 0; i < highlighted_cols.length; i++) {
+    if (highlighted_cols[i].length == 0) continue
+    for (let j = 0; j < highlighted_rows.length; j++) {
+      if (highlighted_rows[j].length == 0) continue
+      const targetNode = new_dict[`${row_nodes[0].name}, ${col_nodes[0].name}`];
 
+      for (let i = 0; i < arrs.mask_arr.length; i++) {
+        if (arrs.position_arr.new[i][0] === targetNode[0]) {
+          arrs.mask_arr[i] += 0.3;
+        }
+        if (arrs.position_arr.new[i][1] === targetNode[1]) {
+          arrs.mask_arr[i] += 0.3;
+        }
```

Comment:
What do `[i][0]` and `[i][1]` mean in this context? I can tell basically that you're checking if the row/col is the same... I think we can just directly set elements in the row/col without checking for it, right -- since we already have `targetNode` which looks to be the row/col of the highlighted node. I just get a bit worried as soon as I see a triply nested loop, and I want to make sure here that we really want to do that. It looks like this should be an O(n) instead of O(n^3) operation.

Reply:
It's complex.
In clustergrammer, the position data and node value data are not directly connected. 

In our case, we can use row node name and col node name to find the specific node with x, y positions. However, it's hard to find the adjacent nodes since they are not continuously stored in our data structure. The only thing we can do is find the intersected node and go through the position value.

Hope this will make sense to you.

Reply:
Is that because the node might have moved due to reordering/reclustering?

Reply:
Yep

Reply:
So maybe we could have a mapping data structure that tells us the new position from the original. That way we could do all these operations in O(1) lookup time

Reply:
This could be the solution. However, we might need to store all nodes with all different rankings. This gonna be a huge memory cost when we have large dataset.